ansible/playbook/README.md

```diff
@@ -1,4 +1,98 @@
 
+# Introduction
+
+This document describes the Ansible implementation in the Snowdrop project.
+
+When in this document it refers to the `controller` it means the machine that executes the Ansible playbooks and roles and that owns the inventory, *i.e.*, the 
+machine that clones this project.
+
+The `hosts` are the machines where the playbooks and roles are executed against. 
+
+The exception goes to the playbooks that are executed against `localhost`. This special host is referred to the `controller`. 
+
+# Ansible Inventory
+
+Ansible inventory manages access to hosts using Ansible.
+
+The inventory is comprised of several files and scripts based at the `k8s-infra/ansible/inventory` folder.
+
+The inventory is built from the `passwordstore` project using the `pass_inventory.py` python script. This task is done automatically when Ansible is executed.
+
+## Groups
+
+Ansible hosts can be grouped into...well groups. This allows the excecution of playbooks and the definition of variables in a common matter for different hosts. 
+
+To assign variables to a group of hosts a file must be created in the  `k8s-infra/ansible/inventory/group_vars` folder. The name of the file must be the same
+as the name of the Ansible group.
+
+So if we want to define the ports that a k8s master should have open we can create a file named `masters` for instance in the `group_vars` folder and add there
+every definition required such as:
+
+```
+firewalld_public_ports:
+  - 6443/tcp
+  - 10250/tcp
+  - 10255/tcp
+  - 8472/udp
+  - 30000-32767/tcp
+```
+
+The assignment of hosts to groups is done in the passwordstore by adding an entry with the name of the host to a folder with the name of the group:
+
+```bash
+$ pass generate ansible/inventory/k8s_115/n01-k115 5
+```
+
+To remove the host from the group.
+
+```bash
+$ pass rm ansible/inventory/k8s_115/h01-k115
+```
+
+# Host management in Ansible
+
+The section describes how to maintain hosts in Ansible and their use. It's execution goes against the `controller`.
+
+## Create a host
+
```

Comment:
We should perhaps mention as pre-req to `git pull` the password project from gitlab - yes/no ? like also how we culd specify another password store path `export PASSWORD_STORE_DIR=~/.password-store-snowdrop` ?

Reply:
That information is on gitlab. I'm not very fond of adding that information here as it is private. But I could add a small reference.